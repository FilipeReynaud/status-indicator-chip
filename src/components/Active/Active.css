.active-div {
	width: 80px;
	height: 25px;
	background-color: #ddecd1;
	border-radius: 50px;
	position: relative;
	margin: auto;
}

.active-circle {
	width: 10px;
	height: 10px;
	background-color: #0eaf00;
	border-radius: 100%;
	position: absolute;
	margin-top: 7.5px;
	margin-left: 7.5px;
	animation-name: active-animation;
	animation-duration: 1s;
	animation-iteration-count: infinite;
}

.active-text {
	color: #0eaf00;
	font-family: "Roboto", sans-serif;
	font-weight: bold;
	font-size: 11px;
	position: absolute;
	right: 10px;
	top: 4px;
}

@keyframes active-animation {
	from {
		background-color: #0eaf00;
	}

	to {
		background-color: transparent;
	}
}
