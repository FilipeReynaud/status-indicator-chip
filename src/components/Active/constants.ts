// Styles
const COLOR = "#0eaf00";
const BACKGROUND_COLOR = "#ddecd1";

// Defaults
const LABEL = "Active";

export { COLOR, BACKGROUND_COLOR, LABEL };
