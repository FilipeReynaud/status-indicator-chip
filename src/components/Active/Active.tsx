import React from "react";

// Components
import BaseChip from "../BaseChip";

// Constants
import * as constants from "./constants";

// Types
export interface ActiveProps {
	label?: string;
	size?: "sm" | "md" | "lg";
}

const Active = (props: ActiveProps) => (
	<BaseChip
		fontColor={constants.COLOR}
		backgroundColor={constants.BACKGROUND_COLOR}
		label={props.label ?? constants.LABEL}
		size={props.size ?? constants.SIZE}
		withBlinkAnimation
	/>
);

export default Active;
