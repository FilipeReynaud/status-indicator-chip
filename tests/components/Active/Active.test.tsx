import React from "react";
import { getByTestId, render, screen } from "@testing-library/react";

// Components
import Active from "../../../src/components/Active";

// Types
import { ActiveProps } from "../../../src/components/Active/Active";

describe("Active", () => {
	it("should render properly", () => {
		// Act
		const { container } = render(<Active />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	it("should have 'active' label", async () => {
		// Arrange
		render(<Active />);

		// Assert
		expect(await screen.findByText("Active")).toBeTruthy();
	});

	it("should be able to set another label", async () => {
		// Arrange
		const defaultLabel = "Active";
		const testLabel = "Dispatched";

		render(<Active label={testLabel} />);

		// Assert
		expect(await screen.queryByText(defaultLabel)).toBeNull();
		expect(await screen.findByText(testLabel)).toBeTruthy();
	});

	it("should have 'active' label with correct color", async () => {
		// Arrange
		const expectedLabelColor = "rgb(14, 175, 0)";
		const { container } = render(<Active />);

		// Assert
		const labelSpan = getByTestId(container, "status-indicator-label");
		expect(labelSpan.style.color).toEqual(expectedLabelColor);
	});

	it("should have background with correct color", async () => {
		// Arrange
		const expectedBackgroundColor = "rgb(221, 236, 209)";
		const { container } = render(<Active />);

		// Assert
		const tableDiv = getByTestId(container, "status-indicator-table");
		expect(tableDiv.style.backgroundColor).toEqual(expectedBackgroundColor);
	});

	it("should have circle with correct color", async () => {
		// Arrange
		const expectedCircleColor = "rgb(14, 175, 0)";
		const { container } = render(<Active />);

		// Assert
		const circleDiv = getByTestId(container, "status-indicator-circle");
		expect(circleDiv.style.backgroundColor).toEqual(expectedCircleColor);
	});

	[
		{ expectedLabelSize: "8px", size: "sm" },
		{ expectedLabelSize: "11px", size: "md" },
		{ expectedLabelSize: "16px", size: "lg" },
	].forEach(({ expectedLabelSize, size }) => {
		it(`should have 'active' label with correct ${size} size`, async () => {
			// Arrange
			const { container } = render(
				<Active size={size as ActiveProps["size"]} />
			);

			// Assert
			const labelSpan = getByTestId(container, "status-indicator-label");
			expect(labelSpan.style.fontSize).toEqual(expectedLabelSize);
		});
	});

	[
		{ expectedCircleSize: "8px", size: "sm" },
		{ expectedCircleSize: "10px", size: "md" },
		{ expectedCircleSize: "12px", size: "lg" },
	].forEach(({ expectedCircleSize, size }) => {
		it(`should have circle div with correct ${size} size`, async () => {
			// Arrange
			const { container } = render(
				<Active size={size as ActiveProps["size"]} />
			);

			// Assert
			const circleDiv = getByTestId(container, "status-indicator-circle");
			expect(circleDiv.style.height).toEqual(expectedCircleSize);
			expect(circleDiv.style.width).toEqual(expectedCircleSize);
		});
	});

	[
		{ expectedPadding: "0px 4px 0px 2px", size: "sm" },
		{ expectedPadding: "0px 7px 0px 5px", size: "md" },
		{ expectedPadding: "0px 10px 0px 8px", size: "lg" },
	].forEach(({ expectedPadding, size }) => {
		it(`should have table with correct ${size} size`, async () => {
			// Arrange
			const { container } = render(
				<Active size={size as ActiveProps["size"]} />
			);

			// Assert
			const tableDiv = getByTestId(container, "status-indicator-table");
			expect(tableDiv.style.padding).toEqual(expectedPadding);
		});
	});

	it("should have circle with correct animation name", async () => {
		// Arrange
		const expectedAnimationName = "blink";
		const { container } = render(<Active />);

		// Assert
		const circleDiv = getByTestId(container, "status-indicator-circle");
		expect(circleDiv.style.animationName).toEqual(expectedAnimationName);
	});

	it("should have circle with correct animation color", async () => {
		// Arrange
		const expectedAnimationColor = "#0eaf00";
		const { container } = render(<Active />);

		// Assert
		const circleDiv = getByTestId(container, "status-indicator-circle");
		const computedStyles = window.getComputedStyle(circleDiv);
		const animationColor =
			computedStyles.getPropertyValue("--animation-color");
		expect(animationColor).toEqual(expectedAnimationColor);
	});
});
