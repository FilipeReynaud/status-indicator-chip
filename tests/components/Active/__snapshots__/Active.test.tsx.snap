// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active should render properly 1`] = `
<div>
  <table
    class="status-indicator-table"
    data-test="status-indicator-table"
    style="padding: 0px 7px 0px 5px; background-color: rgb(221, 236, 209);"
  >
    <tbody>
      <tr>
        <td>
          <div
            class="status-indicator-circle"
            data-test="status-indicator-circle"
            style="width: 10px; height: 10px; background-color: rgb(14, 175, 0); --animation-color: #0eaf00; animation-name: blink; animation-duration: 1s; animation-iteration-count: infinite;"
          />
        </td>
        <td>
          <span
            class="status-indicator-label"
            data-test="status-indicator-label"
            style="color: rgb(14, 175, 0); font-size: 11px;"
          >
            Active
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
